src/os.cc

```diff
@@ -20,6 +20,9 @@
 
 #  ifndef _WIN32
 #    include <unistd.h>
+#    ifndef mode_t
+#      define mode_t unsigned short
+#    endif
```

Comment:
The ``mode_t`` type is defined in the POSIX standard (https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html).

Please, remove.